### ISSUE TITLE
migratewallet: backup file has inconvenient name

### ISSUE THREAD

**Issue by laanwj** [bitcoin/bitcoin#29584]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behaviour

The `migratewallet` RPC can create a backup named (something like) `-998239328.legacy.bak`. Files starting with a `-` are somewhat inconvenient to work with in UNIX, because commands confuse them with a flag. It's usually possible to work around this with `--` but still, it would be more convenient to use positive numbers only.

### Expected behaviour

It would be more convenient to use positive numbers only.

### Steps to reproduce

`bitcoin-cli migratewallet ""`

### Relevant log output

N/A

### How did you obtain Bitcoin Core

Compiled from source

### What version of Bitcoin Core are you using?

v26.0

### Operating system and version

Debian Linux

**Comment by furszy** [comment]
> more convenient to use positive numbers only

It isn't a negative number. It is just a separator. The file format is "<wallet_name>-<current_time>.legacy.bak". But sure, it could easily be replaced for an underscore separator.

**Comment by laanwj** [comment]
Yes, I migrated the default wallet. In that case, instead of changing the seperator, substituting something else instead of "" would work too. It's fine to have `-` in a filename, just not as first character.

**Comment by furszy** [comment]
Done. #29586.
